packages/components/src/menu-item/index.js

```diff
@@ -47,32 +46,26 @@ export function MenuItem( {
 		);
 	}
 
-	let tagName = Button;
-
-	if ( icon ) {
-		if ( ! isString( icon ) ) {
-			icon = cloneElement( icon, {
-				className: 'components-menu-items__item-icon',
-				height: 20,
-				width: 20,
-			} );
-		}
-
-		tagName = IconButton;
-		props.icon = icon;
+	if ( icon && ! isString( icon ) ) {
+		icon = cloneElement( icon, {
+			className: 'components-menu-items__item-icon',
+			height: 20,
+			width: 20,
+		} );
 	}
 
-	return createElement(
-		tagName,
-		{
+	return (
+		<IconButton
```

Comment:
It's a bit stretched use of IconButton in the case when there is no icon 😅 

However, I can't think of another simple way of handling it where this doesn't trigger re-render of the button. I noticed that this behavior is broken only because we swap `IconButton` with `Button` for no reason when only the icon gets removed/added.

Reply:
Yep, exactly! Switching between the two components causes React to remount the component and lose focus. Looking at `IconButton`, it seemed to me that `icon={}` is an optional prop so I felt comfortable doing it this way.

Reply:
One thing we keep discussing here and there is why not just add an `icon` prop to the `Button` block and deprecate the `IconButton` (or make it just a shortcut)

Reply:
I'm sure @drw158 @cburton4 have some good ideas on how to approach it :)

I'm fine with anything which makes it less confusing. I guess having more flexible `Button` and `IconButton` as a backward-compatible alias makes a lot of sense.

Reply:
We have an issue open to discuss combining IconButton and Button in #16541 

We are leaning towards keeping them separate, but I'll look at this closer to understand the disadvantage.

Reply:
I'm assuming from the code that this is an issue because Button does not have an icon option. Switching between icon and non-icon buttons for Menu Items is causing technical problems.

Could this be solved by adding an icon option to the Button component, but still keep IconButton a separate component?

Reply:
To be fair, we already do the opposite here. Not pass the "icon" prop to the IconButton (so it behaves like a Button).  If it doesn't make sense to merge the two components into one, I feel we shouldn't do anything specific about it.